doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,349 @@
+                    }
+                    default "";
+                    description "Set if include the AS set when advertising the aggregated address";
+                }
+            }
+        }
+    }
+}
+```
+
+#### Config DB Sample
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "bbr-required": "true",
+            "summary-only": "false",
+            "aggregate-address-prefix-list": "AGG_ROUTES_V4",
+            "contributing-address-prefix-list": "AGG_CONTRIBUTING_ROUTES_V4"
+        },
+        "fc00::/63": {
+            "bbr-required": "true",
+            "summary-only": "true",
+            "as-set": "true"
+        }
+    }
+    ...
+}
+```
+
+#### Parameters
+We have parameters to control the behavior of aggregate address in bgp container.
+
+##### BBR Required
+It's a boolean to indicate whether only generate aggregated address when BBR feature is enabled.
+
+If it's true and the BBR feature is not enabled, aggregated address won't generated.
+
+##### Summary Only
+It's a boolean to indicate whether only advertise aggregate address only.
+
+If it's true, then contributing routes will be suppressed and only aggregated address will be advertised.
+
+##### AS Set
+It's a boolean to indicate whether add a as set of contributing routes in as path when advertising aggregated address.
+
+If it's true, then when advertising aggregated address there will be a as set of contributing routes appended at as path.
+
+##### Aggregate Address Prefix List
+It's a string which should be the name of a prefix list, and the aggregate address will be append to the prefix list.
+
+##### Contributing Address Prefix List
+It's a string which should be the name of a prefix list, and the aggregate address will be append to the prefix list with prefix length filter to filter prefixes whose length greater or equal to the prefix length of the aggregate address.
+
+### State DB Extension
+For every aggregated address, we track its state in state DB, it has two states active and inactive. Active state means the address is configurated in the bgp container, while inactive state means isn't.
+
+#### State DB sample:
+```json
+{
+    ...
+    "BGP_AGGREGATE_ADDRESS": {
+        "192.168.0.0/24": {
+            "state": "inactive",
+            "aggregate-address-prefix-list": "AGG_ROUTES_V4",
+            "contributing-address-prefix-list": "AGG_CONTRIBUTING_ROUTES_V4"
+        },
+        "fc00::/63": {
+            "state": "active",
+            "aggregate-address-prefix-list": "",
+            "contributing-address-prefix-list": ""
+        }
+    }
+    ...
+}
+```
+
+#### State Transition Diagram
+
+<p align=center>
+<img src="img/aggregate_address_state_transition.png" alt="state-transition">
+</p>
+
+
+### Bgp Container Behavior
+The bgp container will subscribe the keys `BGP_AGGREGATE_ADDRESS` and `BGP_BBR` in config DB. The possible events and related process are:
+1. Add address in config DB:
```

Comment:
If any parameters got updated in config DB, we will remove the address first then add the address back.